### ISSUE TITLE
Open Bio-Formats image files in MoBIE

### ISSUE THREAD

**Issue by tischi** [mobie/mobie-viewer-fiji#921]
@NicoKiaru Is there a simple method in bdv-image-loaders to load a bio-formats file as `SpimData`?

Something like 

```
AbstractSpimData open( String imagePath );
```

I had a quick look but did not see it.

**Comment by tischi** [comment]
Is that it?
https://github.com/BIOP/bigdataviewer-image-loaders/blob/353fa4001a58af6a584058837147d82a50fedd67/src/main/java/ch/epfl/biop/bdv/img/bioformats/command/BdvShowFileBioFormatsCommand.java#L64

**Comment by tischi** [comment]
Why does one need to specify the unit?

**Comment by NicoKiaru** [comment]
Yep that's it. You need to specify the unit because bigdataviewer works unitless (or let's say, it's a global unknown unit). So if the pixel size of your image is a 100 nm, the pixel size should be 100 units of bdv, if bdv is in nanometer, or 0.1 units of bdv, if bdv is in micrometer, of 0.0001 units of bdv, if bdv is in millimeter.

That serves to set the affine transform that is applied to go from pixel to bdv units (and which is store in the ViewRegistration in SpimData)

**Comment by tischi** [comment]
Thanks!

1. If I don't specify it, it will stay in the unit of the image?
2. Should one then also add a `unit` field [here](https://github.com/BIOP/bigdataviewer-image-loaders/blob/353fa4001a58af6a584058837147d82a50fedd67/src/main/java/ch/epfl/biop/bdv/img/imageplus/ImagePlusToSpimData.java#L59)?!

**Comment by NicoKiaru** [comment]
> 1. 

hum I think it'll be micrometer:

https://github.com/BIOP/bigdataviewer-image-loaders/blob/353fa4001a58af6a584058837147d82a50fedd67/src/main/java/ch/epfl/biop/bdv/img/opener/OpenerSettings.java#L78

> 2.

I think it takes the unit of the calibration of the imageplus. The computation is done here I think:

https://github.com/BIOP/bigdataviewer-image-loaders/blob/353fa4001a58af6a584058837147d82a50fedd67/src/main/java/ch/epfl/biop/bdv/img/imageplus/ImagePlusToSpimData.java#L163-L164

**Comment by tischi** [comment]
To me it seems more consistent if

1. _both_ methods would keep the unit of the input data if nothing else is specified
2. _both_ methods would have the option to change the unit

What do you think?

**Comment by tischi** [comment]
I have another question:  could you explain a little bit what the differences between:

1. `OpenersToSpimData.getSpimData(openerSettings);`

and something like 

2. `ImagePlusToSpimData(openAsImagePlusUsingBioFormats(path))`

are? 

Ah, could it be that (1) can also handle multi-resolution file formats, such as, I don't know, maybe CZI?

**Comment by NicoKiaru** [comment]
> 1. _both_ methods would keep the unit of the input data if nothing else is specified

But what is the unit of the input data ? Is there a way to know the 'default unit of the dataset' in bioformats ? I'm not sure such a thing exists.

> 2. both methods would have the option to change the unit

Why not changing the ImagePlus calibration before calling the function ?


> OpenersToSpimData.getSpimData(openerSettings);

This is much more versatile and powerful. This can be used to open a file the BioFormats (multiresolution) API, or OMERO (multiresolution as well), and also to open a QuPath entry, directly from a project file and an entry number.

We needed this to be able to be compatible with QuPath and its various image server.

> ImagePlusToSpimData(openAsImagePlusUsingBioFormats(path))

This won't be multiresolution indeed. And won't cover images bigger than 2Gb per plane.

**Comment by tischi** [comment]
> But what is the unit of the input data ? Is there a way to know the 'default unit of the dataset' in bioformats ? I'm not sure such a thing exists.

Interesting. If you open an image in Fiji using Bio-Formats it has some unit. I don't know where that comes from...
Maybe @joshmoore knows?
